src/viztracer/modules/vcompressor/vc_dump.c

```diff
@@ -134,16 +185,31 @@ int dump_parsed_trace_events(PyObject* trace_events, FILE* fptr)
         uint64_t pid = PyLong_AsLong(PyTuple_GetItem(key, 0));
         uint64_t tid = PyLong_AsLong(PyTuple_GetItem(key, 1));
         const char* name = PyUnicode_AsUTF8(PyTuple_GetItem(key, 2));
-        uint64_t ts_size = PyList_GET_SIZE(value);
+        uint64_t ts_size = PyDict_Size(value);
+        uint64_t ts_cache = 0;
+        PyObject* ts_keys = PyDict_Keys(value);
         fputc(VC_HEADER_FEE, fptr);
         fwrite(&pid, sizeof(uint64_t), 1, fptr);
         fwrite(&tid, sizeof(uint64_t), 1, fptr);
         fwritestr(name, fptr);
         fwrite(&ts_size, sizeof(uint64_t), 1, fptr);
+        if (!ts_keys || !PyList_Check(ts_keys)) {
+            PyErr_SetString(PyExc_ValueError, "failed to get timestamp list");
+            goto clean_exit;
+        }
+        if (PyList_Sort(ts_keys) == -1) {
+            goto clean_exit;
+        }
         for (Py_ssize_t idx = 0; idx < (Py_ssize_t)ts_size; idx++) {
-            double ts = PyFloat_AsDouble(PyList_GET_ITEM(value, idx));
-            int64_t ts64 = ts * 1000;
-            fwrite(&ts64, sizeof(int64_t), 1, fptr);
+            PyObject * ts_obj = PyList_GET_ITEM(ts_keys, idx);
+            double ts = PyFloat_AsDouble(ts_obj);
+            double dur = PyFloat_AsDouble(PyDict_GetItem(value, ts_obj));
+            uint64_t ts64 = ts * 100;
+            uint64_t dur64 = dur * 100;
+            uint64_t delta_ts = ts64 - ts_cache;
+            ts_cache = ts64;
```

Comment:
I prefer not to call this `ts_cache` as it's not a cache. `last_ts` would be a better name.